Turn off scalar_check for _th_normal.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #30502 [BC-BREAKING] change index_select scalar_check to retain dimensionality of input.
* #30438 Turn off scalar_checks for SpatialDepthwiseConvolution and SpatialConvolutionMM.
* #30437 [WIP] Remove scalar_check from topk, move it to the THC implementation.
* #30436 Turn off scalar_checks for _th_reciprocal.
* #30435 Turn off scalar_checks for torch.clamp.
* #30434 Turn off scalar_checks for exp, cos, cosh, tan, atan, tanh, erf, erfc.
* **#29955 Turn off scalar_check for _th_normal.**

Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

gchanan committed:
[WIP] Turn off scalar_check for _th_normal.

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan committed:
Update on "[WIP] Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

albanD reviewed (approved):
Should we still mark this as BC-breaking even though it's very minor?

gchanan committed:
Update on "Turn off scalar_check for _th_normal."


Differential Revision: [D18548051](https://our.internmc.facebook.com/intern/diff/D18548051)

[ghstack-poisoned]

gchanan commented:
@albanD good question.  In this case it's very clearly a bug and CPU did the right thing, so it doesn't seem more BC breaking than any other bug fix that people may rely on.
